TransformationSystem/DB/TransformationDB.py

```diff
@@ -1525,3 +1521,94 @@ def __addDirectory( self, path, force ):
       else:
         successful.append( lfn )
     return {"OK":True, "Value": len( res['Value']['Successful'] ), "Successful":successful, "Failed": failed }
+
+  def setMetadata( self, path, usermetadatadict ):
+    """ It can be applied to a file or to a directory (path). For a file, add the file to Transformations if the updated metadata dictionary passes the filter.
+        For a directory, add the files contained in the directory to the Transformations if the the updated metadata dictionary passes the filter.
+    """
+    gLogger.info( "setMetadata: Attempting to set metadata %s to %s" % (usermetadatadict, path) )
+    successful = {}
+    failed = {}
+    if isinstance( path, dict ):
+      path = path.keys()[0]
+
+    transFiles = {}
+    filesToAdd = []
+
+    catalog = FileCatalog()
+    isFile = catalog.isFile( path )['Value']['Successful'][path]
+    isDirectory = catalog.isDirectory( path )['Value']['Successful'][path]
+
+    if isFile:
+      res = catalog.getFileUserMetadata( path )
+    elif isDirectory:
+      res = catalog.getDirectoryUserMetadata( path )
+
+    if not res['OK']:
+      failed[path] = res['Message']
+      return S_OK( {'Successful':successful, 'Failed':failed } )
```

Comment:
I don't think so, because the Failed/Successful dict are indexed on the input path. Since there is a single one, I don't think it is worth having this extra level